### PR TITLE
MES-6894 Back Button &  Burger Menu Bug

### PR DESCRIPTION
**Pull request by tomcrawleyy**
## Description
Introduces logic to fix the burger menu being scrollable and introduces a custom component to fix bug related to double clicking back button

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [x] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image
image

### CODE REVIEW

File: src/components/common/back-button/back-button.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+:host {
+  .colour-white {
+    color: white;
```
Comment (andrewsetterfield) [inline src/components/common/back-button/back-button.scss]: can we change this to use a variable (mes-white?)

**Comment by andrewsetterfield** [comment]
no spec for new component?

**Comment by andrewsetterfield** [comment]
Text is bigger for 'Back'. Chevron is smaller.  Try these styles 
image

**Comment by tomcrawleyy** [comment]
> no spec for new component?

seems quite pedantic, not sure theres anything really to unit test

**Comment by tomcrawleyy** [comment]
> Text is bigger for 'Back'. Chevron is smaller. Try these styles
> image

will look into this

File: src/components/common/back-button/back-button.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Component } from '@angular/core';
+import { Location } from '@angular/common';
+
+@Component({
+  selector: 'mes-back-button',
+  templateUrl: 'back-button.html',
+  styleUrls: ['./back-button.scss'],
+})
+export class MesBackButtonComponent {
```
Comment (Cullima) [inline src/components/common/back-button/back-button.ts]: can you make the naming consistent, i.e. name the file and folders mes-back-button in keeping with the class and selector?

**Review by Cullima — CHANGES_REQUESTED** [review]

File: src/components/common/mes-back-button/mes-back-button.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  @Input()
+  public routerLink: string;
+  public buttonDisabled = false;
+
+  constructor(
+    private router: Router,
+  ) { }
+
+  public async onClick(event: Event) {
```
Comment (andrewsetterfield) [inline src/components/common/mes-back-button/mes-back-button.ts]: 'promise<void>' as return value?

Comment (andrewsetterfield) [reply src/components/common/mes-back-button/mes-back-button.ts]: hmm, apparently i cant write promise < void >

Comment (tomcrawleyy) [reply src/components/common/mes-back-button/mes-back-button.ts]: not sure what you mean sorry

**Review by andrewsetterfield — APPROVED** [review]

**Review by Cullima — APPROVED** [review]

**Pull request merged**